### ISSUE TITLE
Thunderbird UI obstructed by history popup

### ISSUE THREAD

**Issue by jenda69** [opto/Expression-Search-NG#120]
Every time I click the Expression Search field, it pops up really wide history window which covers the top of the Thunderbird UI and the search field itself.

Once I start typing, it disappears, but it is annoying because if I try to change caret position by mouse or try to click (×) to clear the field, the history pops up again instead and I just can't do the thing I want to do.
Secondly, it also covers a lot of Thunderbird UI except the tabs which are actually shown over the pop up, so it's really ugly. 

I managed to somewhat solve the first issue by adding bunch of <space> searches to the history, now the window got quite narrow and at least doesn't cover the search field...

![image](https://github.com/opto/Expression-Search-NG/assets/932875/17f04b5e-fbeb-4a40-9852-32d5107cac88)

Is there way to turn this off? I don't really need such feature anyway.

**Comment by avegaweiss** [comment]
hi! Look at the comments on issue #119 . I've made a version without this issue (but other problems...)
I'm using it, and I am almost happy.

**Comment by serjsk** [comment]
Would be nice to have this history optional. I personally don't need it at all...

**Comment by darwin4ever** [comment]
Very annoying that "History Popup"

I don't need it either, so it would be great if
- either that popup could be made optional with a separate option
- or the whole history made optional

The good news : "simple:" is working again in 4.0.44

**Comment by mooredj19** [comment]
Agreed:  so irritating that I've disabled this addon -- especially since it doesn't even work for me (I cannot click the box)
It hit me 2024Nov01.  It took a bit of troubleshooting to figure out what addon was so rude.
I"m hoping this gets fixed, so I can re-enable this addon.

**Comment by skypal** [comment]
Hi everybody, 
I do like this plugin very much (kudos to the creators/maintainers). But I was like "whooah, what the *** is that?" when this screen popped up the first time. I must agree that this screen is very annoying and I am surprised that there is no "opt-out" in the prefs to disable this. 
To show a dialog like this feels like "premature obedience": the plugin intends to knows better what I want to do next -- which is obviously not the case for some of us. Please fix this somehow or better come up with something different to show the history. In i.e. Sublime Editor, there is a popup that lists previous querys...

**Comment by avegaweiss** [comment]
opto: many thanks sfor your work!
But this history-popup-thing is kind of annoying.
I've commented out a line, and I thinks this feature isn't enable anymore....which is fine for me.
Sharing for anyone to test.


[expression-search-NG4.0.40.mod.xpi.zip](https://github.com/user-attachments/files/17671462/expression-search-NG4.0.40.mod.xpi.zip)

**Comment by carlosfhr** [comment]
I love this addon and i feel really grateful for it, but please remove the search history popup is really really annoying. At lest add an option to disable it.

![imagen](https://github.com/user-attachments/assets/84d8e36b-0927-4567-938e-05f7b1514df7)

**Comment by Fulvus** [comment]
A peculiar quirk is that it's only showing up if Thunderbird is full screen.
Would be nice to be able to turn it off or at least move its postion to similar of the Search Tip text box.

**Comment by Inserm-CRI-LBI** [comment]
> opto: many thanks sfor your work! But this history-popup-thing is kind of annoying. I've commented out a line, and I thinks this feature isn't enable anymore....which is fine for me. Sharing for anyone to test.
> 
> [expression-search-NG4.0.40.mod.xpi.zip](https://github.com/user-attachments/files/17671462/expression-search-NG4.0.40.mod.xpi.zip)

Same here, specifically line 789 in /modules/ExpressionSearchChrome.jsm 
//    ExpressionSearchChrome.createHistoryHTMLDialog(win);
seems to be the "culprit". I commented it and the popup goes away (quite unclean solution as the underlying function is still there, but I may reuse it later on who knows). It may be useful to others, so I think it would be great to add the possibility to enable or disable it in the options panel. 
Anyways, thanks for this awesome and essential addon.

**Comment by darwin4ever** [comment]
> A peculiar quirk is that it's only showing up if Thunderbird is full screen. Would be nice to be able to turn it off or at least move its postion to similar of the Search Tip text box.

I'm not using TB full screen ... but have it showing up

**Comment by Joymaker** [comment]
I have the problem (on Mac) even if TB is NOT full screen.

**Comment by opto** [comment]
see https://github.com/opto/Expression-Search-NG/releases/tag/v4.1.45

either set a position in settings or disable it

**Comment by Alfamari** [comment]
> see https://github.com/opto/Expression-Search-NG/releases/tag/v4.1.45
> 
> either set a position in settings or disable it

I do not see any settings on how to move or disable it.

**Comment by opto** [comment]
click the search glass in the status line, choose settings, ca. the 8th checkbox from the top, press ok on that window.

**Comment by Alfamari** [comment]
@opto I'm sorry for being so late I got busy and then forgot to follow-up, but there aren't 8 check boxes in the top section.

![Image](https://github.com/user-attachments/assets/de02d3fb-c7af-4ec0-b647-ad6aea706110)

EDIT: I've found the reason is because my client cannot update the extension. I am using betterbird with scoop so I will try fixing this.

Second Edit: Looks like its just a betterbird version compatibility issue that I'll have to wait for an update from them.
